cmd/neofs-adm/internal/modules/morph/deploy.go

```diff
@@ -95,9 +98,32 @@ func deployContractCmd(cmd *cobra.Command, args []string) error {
 			cs.Manifest.Name)
 	}
 
-	err = c.addManifestGroup(cs.Hash, cs)
-	if err != nil {
-		return fmt.Errorf("can't sign manifest group: %v", err)
+	force, _ := cmd.Flags().GetBool(forceFlag)
+
+	if c.ContractWallet == nil {
+		if !force {
+			return fmt.Errorf("`%s` is not found, use `--%s` to deploy without group",
+				contractWalletFilename, forceFlag)
+		}
+	} else {
+		pub := c.ContractWallet.Accounts[0].PrivateKey().PublicKey()
+		group, err := nnsResolveKey(c.Client, nnsCs.Hash, morphclient.NNSGroupKeyName)
+		if err != nil {
+			return fmt.Errorf("can't resolve group key via NNS: %w", err)
+		}
+
+		if !pub.Equal(group) && !force {
+			return fmt.Errorf(
+				"`%s` does not equal NNS's `%s`; use `--%s` flag to deploy with that key anyway",
+				contractWalletFilename, morphclient.NNSGroupKeyName, forceFlag,
```

Comment:
Maybe some "%s doesn't contain NNS's group key..." will be better? The first argument is wallet filename, this message looks strange a bit.

Reply:
decided not to deploy custom contracts with `group.neofs` group, removed that code